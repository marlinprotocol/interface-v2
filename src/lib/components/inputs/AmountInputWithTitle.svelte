<script lang="ts">
	import InputCard from '$lib/atoms/cards/InputCard.svelte';
	import { inputClasses } from '$lib/atoms/componentClasses';
	import Text from '$lib/atoms/texts/Text.svelte';
	import AmountInput from '$lib/components/inputs/AmountInput.svelte';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let title: string;
	export let disabled = false;
	export let inputAmountString = '';
	export let handleUpdatedAmount = (e: Event) => {};
	export let prefix = '';
	export let suffix = '';
	export let onlyInteger = false;
	export let id = '';
	export let styleClass = '';
	export let placeholder = '';
</script>

<InputCard variant="search" styleClass="border border-grey-100 relative">
	{#if title}
		<p
			class="absolute left-7 top-[-1px] z-[1] inline-flex bg-base-100 font-poppins text-sm font-normal leading-[1px] text-grey-800"
		>
			{title}
		</p>
	{/if}

	<div class="mt-1 flex items-center">
		{#if !!prefix.length}
			<Text text={prefix} styleClass="text-sm text-grey-800" />
		{/if}

		<AmountInput
			{id}
			bind:value={inputAmountString}
			onChange={handleUpdatedAmount}
			styleClass={cn(inputClasses.inputText, styleClass)}
			{onlyInteger}
			{disabled}
			{placeholder}
		/>
		{#if !!suffix.length}
			<Text text={suffix} styleClass="text-sm text-primary ml-1 " />
		{/if}

		<div class="w-fit">
			<slot name="endButton" />
		</div>
	</div>
</InputCard>
