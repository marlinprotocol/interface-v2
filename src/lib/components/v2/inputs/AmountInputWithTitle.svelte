<script lang="ts">
	import InputCard from '$lib/atoms/v2/cards/InputCard.svelte';
	import Text from '$lib/atoms/v2/texts/Text.svelte';
	import AmountInput from '$lib/components/v2/inputs/AmountInput.svelte';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let title: string;
	export let disabled = false;
	export let inputAmountString = '';
	export let handleUpdatedAmount = (e: Event) => {};
	export let prefix = '';
	export let suffix = '';
	export let onlyInteger = false;
	export let id = '';

	const onFocusStyle = 'left-[12px] leading-[1px] top-[-20px] font-normal';
	const titleStyleInit =
		'absolute cursor-text bg-white font-poppins text-sm text-[#030115] transition-all';

	let titleStyle = onFocusStyle;

	const handleChange = (e: any) => {
		handleUpdatedAmount(e);
	};
</script>

<InputCard variant="search" styleClass="border border-[#D9DADE]">
	<div class="relative flex flex-1 flex-col">
		<button class={cn(titleStyle, titleStyleInit)} id="title-{id}">
			{title}
		</button>
	</div>
	<div class="flex items-center">
		{#if !!prefix.length}
			<Text text={prefix} styleClass="text-sm text-[#030115] mt-1" />
		{/if}

		<AmountInput
			{id}
			bind:value={inputAmountString}
			onChange={handleChange}
			styleClass="input input-ghost min-w-[20px] w-full input-primary text-sm p-0 ml-0.5 placeholder:text-primary/[.2] h-[30px] mt-1"
			{onlyInteger}
			{disabled}
		/>
		{#if !!suffix.length}
			<Text text={suffix} styleClass="text-sm text-primary ml-1 " />
		{/if}

		<div class="w-fit">
			<slot name="endButton" />
		</div>
	</div>
</InputCard>
