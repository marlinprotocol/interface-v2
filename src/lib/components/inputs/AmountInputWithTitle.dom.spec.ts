import { render, fireEvent } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';
import AmountInputWithTitle from './AmountInputWithTitle.svelte';

describe('AmountInputWithTitle', () => {
	const title = 'Sample Title';
	const inputAmountString = '123';
	const prefix = '$';
	const suffix = 'USD';
	const onlyInteger = false;
	const handleUpdatedAmount = vi.fn();
	const onFocusOut = vi.fn();

	it("render's the component properly", () => {
		const { container } = render(AmountInputWithTitle, {
			title: title,
			inputAmountString: inputAmountString,
			handleUpdatedAmount: handleUpdatedAmount,
			prefix: prefix,
			suffix: suffix,
			onlyInteger: onlyInteger,
			disabled: false
		});
		expect(container).toMatchSnapshot();
	});

	it('renders the title and input with the passed props', async () => {
		const { findByText, container } = render(AmountInputWithTitle, {
			title: title,
			inputAmountString: inputAmountString,
			handleUpdatedAmount: handleUpdatedAmount,
			prefix: prefix,
			suffix: suffix,
			onlyInteger: onlyInteger,
			disabled: false
		});

		// Assert the title is rendered correctly
		expect(await findByText(title)).toBeTruthy();

		// Assert the prefix and suffix are rendered correctly
		expect(await findByText(prefix)).toBeTruthy();
		expect(await findByText(suffix)).toBeTruthy();

		const input = Array.from(container.getElementsByTagName('input'))[0];

		expect(input.value).toBe(inputAmountString);

		// // Simulate changing the input value
		fireEvent.input(input, { target: { value: '456' } });

		expect(handleUpdatedAmount).toHaveBeenCalled();
		expect(input.value).toBe('456');

		// Simulate the focusout event
		fireEvent.focusOut(input);
		expect(onFocusOut).toHaveBeenCalled();
	});
});
