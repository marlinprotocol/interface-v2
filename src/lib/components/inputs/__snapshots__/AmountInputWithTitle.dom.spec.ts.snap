// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AmountInputWithTitle > render's the component properly 1`] = `
<body>
  <div>
    <div
      class="w-full bg-white px-4 py-2.5 rounded-[100px] border border-[#D9DADE]"
      data-testid="input-card"
    >
      <div
        class="relative flex flex-1 flex-col"
      >
        <button
          class="left-[12px] top-[-20px] font-normal absolute cursor-text bg-white font-poppins text-sm text-[#030115] transition-all"
          id="title-"
        >
          Sample Title
        </button>
      </div>
       
      <div
        class="mt-1 flex items-center"
      >
        <div
          class="text-sm text-[#030115] text-base font-normal"
          data-testid="text"
        >
          $
        </div>
        <!--&lt;Text&gt;-->
         
        <input
          autocomplete="off"
          class="hideInputNumberAppearance input input-ghost min-w-[20px] w-full input-primary text-sm p-0 ml-0.5 placeholder:text-primary/[.2] h-[30px]"
          id="pond-input-amount-"
          placeholder="0.00"
        />
        <!--&lt;AmountInput&gt;-->
         
        <div
          class="text-sm text-primary ml-1  text-base font-normal"
          data-testid="text"
        >
          USD
        </div>
        <!--&lt;Text&gt;-->
         
        <div
          class="w-fit"
        />
      </div>
    </div>
    <!--&lt;InputCard&gt;-->
    <!--&lt;AmountInputWithTitle&gt;-->
  </div>
</body>
`;
